### PR TITLE
Add 1.4.2 changelog

### PR DESCRIPTION
**Pull request by techknowlogick**
As title

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ This changelog goes through all the changes that have been made in each release
...
+  * fix webhook bug of response info is not displayed in UI (#4023)
+  * fix writer cannot read bare repo guide (#4033) (#4039)
+  * Don't force due date to current time (#3830) (#4057)
+  * Fix wiki redirects (#3919) (#4065)
+  * fix attachment ENABLED (#4064) (#4066)
+  * Added deletion of an empty line at the end of file (#4054) (#4074)
+  * Use ResolveReference instead of path.Join (#4073)
+  * Fix #4081 Check for leading / in base before removing it (#4083)
+* MISC
+  * Update git vendor (#4059) (#4075)
```
Comment (lafriks) [inline CHANGELOG.md]: This should be moved under BUGFIXES with description `Respository's home page not updated after first push`

Also please upper case first letter in each sentence.

**Review by lafriks — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (techknowlogick) [reply CHANGELOG.md]: Thanks for feedback @lafriks I've updated changelog accordingly.

File: main.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 )
 
 // Version holds the current Gitea version
-var Version = "1.4.0-dev"
+var Version = "1.4.2"
```
Comment (lafriks) [inline main.go]: No need to change this version will be read from git tag on this branch

**Comment by techknowlogick** [comment]
@lafriks thanks. changed.

**Review by lafriks — APPROVED** [review]

**Review by daviian — APPROVED** [review]

**Comment by lafriks** [comment]
@techknowlogick please backport to master

**Pull request merged**